SIPS/sip-12.md

```diff
@@ -51,12 +51,18 @@ This permission is specified as follows in `snap.manifest.json` files:
   "initialPermissions": {
     "endowment:name-lookup": {
         "chains": ["eip155:1", "bip122:000000000019d6689c085ae165831e93"],
+        "matchers": { "tlds": ["lens"], "nonTldPrefixes": ["farcaster"] }
     }
   }
 }
 ```
 
-`chains` - An array of CAIP-2 chain IDs that the snap supports. This field is useful for a client in order to avoid unnecessary overhead.
+`chains` - A non-empty array of CAIP-2 chain IDs that the snap supports. This field is useful for a client in order to avoid unnecessary overhead.
+`matchers` - A non-empty object that MUST contain 1 or both of the below properties. These matchers are useful for a client for validating input for domain resolution, also helpful in reducing overhead. 
+  `tlds` - An optional non-empty array of top level domains that the snap will provide resolution for.
+  `nonTldPrefixes` - An optional non-empty array of prefixes that the snap expects for non-tld domain lookup.
```

Comment:
I feel a better naming could be used here.

Both are valid [URIs](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Syntax), with the second using `scheme` property of the URI.

```suggestion
  `scheme` - An optional non-empty array of prefixes that the snap expects for non-tld domain lookup.
```

Reply:
Good point, although I will opt to name it `schemes` as the snap can potentially resolve for multiple schemes